internal/command/utils/lifecycle/before.go

```diff
@@ -14,66 +17,102 @@ import (
 	"github.com/valist-io/valist/internal/prompt"
 )
 
-func SetupClient(c *cli.Context) error {
+type contextKey string
+
+const passphraseKey contextKey = "passphrase"
+
+func loadConfig() (*config.Config, error) {
 	home, err := os.UserHomeDir()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	if err := config.Initialize(home); err != nil {
-		return err
+		return nil, err
 	}
 
 	cfg := config.NewConfig(home)
+
 	if err := cfg.Load(); err != nil {
-		return err
+		return nil, err
 	}
 
+	return cfg, nil
+}
+
+func getAccount(c *cli.Context, cfg *config.Config) (accounts.Account, error) {
 	var account accounts.Account
 	if c.IsSet("account") {
 		account.Address = common.HexToAddress(c.String("account"))
 	} else {
 		account.Address = cfg.Accounts.Default
 	}
 
-	client, err := core.NewClient(c.Context, cfg, account)
-	if err != nil {
-		return err
+	// use env variable VALIST_SIGNER as signing key when set
+	if os.Getenv("VALIST_SIGNER") != "" {
+		// generate cryptographically secure, ephemeral encryption key that will clear when program halts
+		randBytes := make([]byte, 32)
+		_, err := rand.Read(randBytes)
+		if err != nil {
+			return accounts.Account{}, err
+		}
+
+		passphrase := fmt.Sprintf("%x", randBytes)
+
+		private, err := crypto.HexToECDSA(os.Getenv("VALIST_SIGNER"))
+		if err != nil {
+			return accounts.Account{}, err
+		}
+
+		// when VALIST_SIGNER is set, this encrypted key is stored in an os.TempDir()
+		// when program halts, the ephemeral encryption key is cleared, making the keystore file inaccessible
+		// this prevents having to clear the keystore file during runtime, which could fail to execute if program is interrupted/crashes
+		// @TODO emulate filesystem in-memory when using VALIST_SIGNER to remove need for this workaround
+		account, err = cfg.KeyStore().ImportECDSA(private, passphrase)
+		if err != nil {
+			return accounts.Account{}, err
+		}
+
+		c.Context = context.WithValue(c.Context, passphraseKey, passphrase)
 	}
 
-	c.Context = context.WithValue(c.Context, core.ClientKey, client)
-	c.Context = context.WithValue(c.Context, core.ConfigKey, cfg)
-	return nil
+	return account, nil
 }
 
-func SetupClientWithPassphrase(c *cli.Context) error {
-	home, err := os.UserHomeDir()
-	if err != nil {
-		return err
-	}
+func UnlockAccount(c *cli.Context) error {
+	client := c.Context.Value(core.ClientKey).(*client.Client)
 
-	if err := config.Initialize(home); err != nil {
-		return err
+	// check for ephemeral encryption key (will be set if using VALIST_SIGNER env var)
+	passphrase := c.Context.Value(passphraseKey).(string)
```

Comment:
This will panic if passphraseKey is not set.
See https://tour.golang.org/methods/15

Reply:
great catch, thanks! added 'ok' check